delta/kernel/src/main/scala/ch/epfl/bluebrain/nexus/delta/kernel/jwt/ParsedToken.scala

```diff
@@ -18,7 +25,21 @@ final case class ParsedToken private (
     expirationTime: Instant,
     groups: Option[Set[String]],
     jwtToken: SignedJWT
-)
+) {
+
+  def validate(audiences: Option[NonEmptySet[String]], keySet: JWKSet): Either[InvalidAccessToken, JWTClaimsSet] = {
+    val proc        = new DefaultJWTProcessor[SecurityContext]
+    val keySelector = new JWSVerificationKeySelector(JWSAlgorithm.RS256, new ImmutableJWKSet[SecurityContext](keySet))
+    proc.setJWSKeySelector(keySelector)
+    audiences.foreach { aud =>
+      proc.setJWTClaimsSetVerifier(new DefaultJWTClaimsVerifier(aud.toSet.asJava, null, null, null))
+    }
+    Either
+      .catchNonFatal(proc.process(jwtToken, null))
+      .leftMap(err => InvalidAccessToken(subject, issuer, err.getMessage))
+  }
```

Comment:
I'm expecting there might be complication with this, but it might be better if we has a constructor that returned an already validated token? There is something about validate being a method you can optionally call which feels a bit weird. Not a big deal though since you've got testing to cover

Reply:
It comes from the way we do things in `IdentitiesImpl`, we do some surface checks first and extract the issuer to:
- Get the matching realm and the public keys associated to it
- Validate the token with the keys
I will check if there is a way to improve that a bit

Reply:
One other thing: if we don't use the `JWTClaimsSet` then it's probably clearer to return `Unit`